lib/Parser/Parse.cpp

```diff
@@ -8437,6 +8450,46 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
     return pnode;
 }
 
+template<bool buildAST>
+void Parser::TrackAssignment(ParseNodePtr pnodeT, IdentToken* pToken, charcount_t ichMin, charcount_t ichLim)
+{
+    if (buildAST)
+    {
+        Assert(pnodeT != NULL);
+        if (pnodeT->nop == knopName)
+        {
+            PidRefStack *ref = pnodeT->sxPid.pid->GetTopRef();
+            Assert(ref);
+            ref->TrackAssignment(pnodeT->ichMin, pnodeT->ichLim);
+        }
+    }
+    else
+    {
+        Assert(pToken != NULL);
+        if (pToken->tk == tkID)
+        {
+            PidRefStack *ref = pToken->pid->GetTopRef();
+            Assert(ref);
+            ref->TrackAssignment(ichMin, ichLim);
+        }
+    }
+}
+
+void PidRefStack::TrackAssignment(charcount_t ichMin, charcount_t ichLim)
+{
+    if (this->isAsg)
+    {
+        if (this->GetIchMin() <= ichMin)
```

Comment:
Sure. WIll have a word with Paul and make necessary changes.

---
In reply to: [67786544](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67786544) [](ancestors = 67786544)